packages/eslint-plugin/src/rules/prefer-readonly-parameter-types.ts

```diff
@@ -94,8 +107,9 @@ export default util.createRule<Options, MessageIds>({
 
           const tsNode = esTreeNodeToTSNodeMap.get(actualParam);
           const type = checker.getTypeAtLocation(tsNode);
-          const isReadOnly = util.isTypeReadonly(checker, type, {
+          const isReadOnly = util.isTypeReadonly(program, type, {
             treatMethodsAsReadonly: treatMethodsAsReadonly!,
+            allowlist: allowlist,
```

Comment:
Nit: a little surprised we don't lint for this in the default internal lint rules 🤔 

```suggestion
            allowlist,
```